yapapi/runner/__init__.py

```diff
@@ -204,6 +204,36 @@ def on_work_done(task, status):
         workers: Set[asyncio.Task[None]] = set()
         last_wid = 0
 
+        agreements_to_pay: Set[str] = set()
+        invoices: Dict[str, rest.payment.Invoice] = dict()
+        payment_closing: bool = False
+
+        async def process_invoices():
+            assert self._budget_allocation
+            allocation: rest.payment.Allocation = self._budget_allocation
+            async for invoice in self._payment_api.incoming_invoices():
+                if invoice.agreement_id in agreements_to_pay:
+                    agreements_to_pay.remove(invoice.agreement_id)
+                    await invoice.accept(amount=invoice.amount, allocation=allocation)
+                else:
```

Comment:
here should be some log output stating that we've received the invoice but not paying it yet